h2/src/main/org/h2/table/TableFilter.java

```diff
@@ -1056,6 +1065,30 @@ public void setAlias(String alias) {
         this.alias = alias;
     }
 
+    /**
+     * Set derived column list.
+     *
+     * @param derivedColumnNames names of derived columns
+     */
+    public void setDerivedColumns(ArrayList<String> derivedColumnNames) {
+        Column[] columns = getColumns();
+        int count = columns.length;
+        if (count != derivedColumnNames.size()) {
+            throw DbException.get(ErrorCode.COLUMN_COUNT_DOES_NOT_MATCH);
+        }
+        HashMap<Column, String> map = new HashMap<>(count);
+        for (int i = 0; i < count; i++) {
+            String alias = derivedColumnNames.get(i);
+            for (int j = 0; j < i; j++) {
```

Comment:
I might be slow today, but shouldn't this loop be:
    for (int j=i+1; j < count; j++) {
?
Otherwise this throw should always occur, because the [0] element will be compared to the [0] element.

Reply:
Why?
```
First iteration:
i = 0;
j = 0;
j(0) < i(0) == false and inner loop is not executed
```